Gruntfile.js

```diff
@@ -559,52 +589,55 @@ module.exports = function (grunt) {
 	/* -------------
 		SERVE
 	------------- */
-	//most basic serve task, it's the safest, but also the slowest.
-	grunt.registerTask('serve', 'Serve the files. --no-test to skip tests', function () {
-		//default --test=true
-		if (typeof grunt.option('test') === "undefined") {
-			grunt.option('test', true);
-		}
+	// default serve task that runs tests and builds and tests dist by default.
+	grunt.registerTask('serve', 'Test, build, serve files. (~20s)', function () {
+		var tasks = ['test', 'servedist'];
+		grunt.task.run(tasks);
+	});
 
-		grunt.task.run(['servedist']);
+	// default serve task that runs tests and builds and tests dist by default (~20s).
+	grunt.registerTask('serveslow', 'Serve files. Run all tests. Does not build. (~20s)', function () {
+		var tasks = ['connect:server', 'test', 'watch:source'];
+		grunt.task.run(tasks);
 	});
 
-	//Fastest serve command for freely slinging code (you won't get interrupted by tests, etc).
-	grunt.registerTask('servefast', 'Serve the files. pass --test to run tests.', function () {
+	//Fastest serve command for freely slinging code (no tests will run by default).
+	grunt.registerTask('servefast', 'Serve the files (no watch), --test to run tests. (~0s)', function () {
 		grunt.task.run(['connect:server']);
 
 		if (grunt.option('test')) {
-			grunt.task.run(['qunit:simple', 'watch:simple']);
+			grunt.task.run(['connect:testServer', 'qunit:source', 'watch:source']);
 		} else {
 			grunt.task.run(['watch:lite']);
 		}
 	});
 
-	//Fastest serve command when you're working on LESS
-	grunt.registerTask('serveless', 'Compile LESS and serve the files. pass --tests to run test.', function () {
+	// Fastest serve command when you're working on LESS
+	grunt.registerTask('serveless', 'Compile LESS and serve the files. pass --tests to run test. (~3s)', function () {
 		grunt.task.run(['distcss']);
 
 		if (grunt.option('test')) {
-			//add qunit:simple as a watch task for watch:less since they want tests
+			// add qunit:source as a watch task for watch:less since they want tests
 			grunt.config.merge({
 				watch: {
 					less: {
-						tasks: ['qunit:simple']
+						tasks: ['qunit:source']
 					}
 				}
 			});
-			grunt.task.run(['qunit:simple']);
+			grunt.task.run(['qunit:source']);
 		}
 
 		grunt.task.run(['connect:server', 'watch:less']);
 	});
 
-	//basically just `grunt serve` but tests default to being off
-	grunt.registerTask('servedist', 'Compile and serve everything, pass --test to run tests.', function () {
+	// same as `grunt serve` but tests default to being off
+	grunt.registerTask('servedist', 'Compile and serve everything, pass --test to run tests. (~7s)', function () {
 		grunt.task.run(['dist']);
 
-		//start up the server here so we can run tests if appropriate
+		//start up the servers here so we can run tests if appropriate
 		grunt.task.run(['connect:server']);
+		grunt.task.run(['connect:testServer']);
```

Comment:
Why start two servers? Which one is supposed to be the one the files are serving from for development? Will both serve all the same files?